src/components/index.js

```diff
@@ -0,0 +1,16 @@
+import template from './index.pug'
+import data from '../../data/data.json'
+import './index.scss'
+
+export class Index {
+  constructor (node) {
+    this.node = node
+    this.insertNode(this.node)
+  }
+
+  insertNode (node) {
```

Comment:
I would change the name of this method since it is not inserting nodes, but instead is inserting a string of html.